screenpipe-server/src/pipe_manager.rs

```diff
@@ -298,6 +298,66 @@ impl PipeManager {
         }
     }
 
+    #[cfg(unix)]
+    pub fn terminate_process_tree(pid: i32) {
+        use nix::sys::signal::{killpg, Signal};
+        use nix::unistd::Pid;
+
+        if let Err(e) = killpg(Pid::from_raw(pid), Signal::SIGKILL) {
+            warn!("Failed to kill process group {}: {}", pid, e);
+        }
+    }
+
+    #[cfg(windows)]
+    pub fn terminate_process_tree(pid: u32) {
+        use windows::Win32::System::JobObjects::*;
+        use windows::Win32::System::Threading::*;
+        use windows::Win32::Foundation::*;
+        
+        unsafe {
+            let h_job = CreateJobObjectA(None, None);
+            if h_job.0 == 0 {
+                warn!("Failed to create job object");
+                return;
+            }
+
+            let mut info = JOBOBJECT_BASIC_LIMIT_INFORMATION {
+                LimitFlags: JOB_OBJECT_LIMIT_KILL_ON_JOB_CLOSE,
+                ..Default::default()
+            };
+            
+            let mut extended_info = JOBOBJECT_EXTENDED_LIMIT_INFORMATION {
+                BasicLimitInformation: info,
+                ..Default::default()
+            };
+
+            if AssignProcessToJobObject(h_job, OpenProcess(PROCESS_ALL_ACCESS, false, pid).unwrap()).is_err() {
+                warn!("Failed to assign process to job object");
+                return;
+            }
+
+            // Closing the job object ensures all child processes are killed
+            CloseHandle(h_job);
+        }
+    }
+    #[cfg(windows)]
+    pub async fn remove_dir_with_retries(path: &Path) -> Result<()> {
```

Comment:
this is wrong

stop != delete

Reply:
Yeah sure we are to stop the pipe and not just delete the directory  associated  with the pipe that why there is a terminate_process _tree function to kill the process  and it's  children. Let me try testing on windows  platform....